csrc/python_frontend/fusion_record.h

```diff
@@ -379,6 +380,78 @@ struct ReshapeOpRecord : RecordFunctor {
   std::vector<int64_t> new_shape_;
 };
 
+struct PadOpRecord : RecordFunctor {
+  PadOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      std::vector<State>& pad_widths)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.pad",
+            RecordType::PadOp),
+        pad_widths_(std::move(pad_widths)) {}
```

Comment:
nitpick. let's check on length of pads & args[0].ndim for semantics.

Reply:
I do that check in `python_bindings.cpp`. I don't think there's any dimension info in a `State` object, so we can't do stuff like that on the `RecordFunctor` side.